cluster/charts/crossplane/templates/deployment.yaml

```diff
@@ -27,8 +27,9 @@ spec:
       serviceAccountName: {{ template "name" . }}
       containers:
       - image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
-        {{- if .Values.args }}
         args:
+        - --leader-election={{ .Values.leaderElection }}
+        {{- if .Values.args }}
         {{- range $arg := .Values.args }}
         - {{ $arg }}
         {{- end }}
```

Comment:
Any idea what happens if someone manually provided `--leader-election` as an arg as well?

Reply:
That I believe will be handled by `kingpin`, not entirely sure yet. I mean errors out or being able to handle duplicate occurrence of a flag. Trying to replicate now.

Reply:
So it did error out. `main: error: flag 'leader-election' cannot be repeated, try --help`

Reply:
Moving this into deployment's `env` could solve but, but I've noticed another thing. It appears kingpin is not able to process negative boolean flags, i.e. `--leader-election=false` doesn't work, but `export LEADER_ELECTION=false` does.

Reply:
Moved this to environment variable of deployment which prevents collision if `--leader-election` was explicitly provided in `args`. Note that explicit arg will override the environment variable, so the intention of the user will be preserved.

Reply:
> It appears kingpin is not able to process negative boolean flags, i.e. --leader-election=false doesn't work, but export LEADER_ELECTION=false does.

I vaguely recall that `kingpin` handles this by using a `--no` prefix, i.e. `--no-leader-election`.

Reply:
Ah interesting, I wasn't aware of `--no` convention. But anyways I reversed the logic for `--leader-election` to be `false` by default and gets enabled by choice.